### ISSUE TITLE
Treat `withPayload` change as new channel

### ISSUE THREAD

**Issue by dustinfarris** [saschatimme/elm-phoenix#25]
I have a scenario where I provide a payload when joining a channel, and the channel replies with a data structure formatted to my specification.

E.g.

```elm
Channel.init channelName
    |> Channel.withPayload (object [ ( "scope", string "full" ) ])
```

So there are 3 situations to consider

1. User enters the application—the first route they see requires channel with a certain payload
2. User enters a route where the channel is not required, then transitions to another route that adds the channel with a certain payload
3. User enters a route, the channel is added with a certain payload, then the user transitions to another route which wants the same channel but with a different payload (a different response from join is desired)

1 and 2 work fine.  3 does not.  Once the channel is added to elm-phoenix channels, changes to `withPayload` do not have any effect.

I realize I could wire up a channel message to verify/adjust parameters on route change (or whatever other model changes) for an existing channel, but I really like the simplicity of just updating a single list of Channels in `Phoenix.connect`—it's one of my favorite things about the elm-phoenix API.

**tl;dr**

I'd like changes to `Channel.withPayload` to be treated as a new channel, dropping the old channel of the same topic (if it exists) and joining the new channel with the new payload.

So is this something the community would be open to?

**Comment by saschatimme** [comment]
I think this would cause some unexpected side effects (e.g. what happens with messages that are not yet send to server ).
Couldn't the same be archived without much overhead if you end `Phoenix.push` `Cmd` after the route transition?

**Comment by dustinfarris** [comment]
Yeah, I can do it with Phoenix.push, just requires a little more effort on the backend to handle it.

I wasn't sure if this was a great idea or not, just seemed like it could be a nice shortcut to handle my situation.

I still think treating `withPayload` as an element of channel uniqueness could be discussed further, but I will use push for now.
